src/coreclr/jit/lclvars.cpp

```diff
@@ -4075,6 +4077,35 @@ void Compiler::lvaMarkLclRefs(GenTree* tree, BasicBlock* block, Statement* stmt,
                 BlockSetOps::AddElemD(this, varDsc->lvRefBlks, block->bbNum);
             }
         }
+
+        if (!varDsc->lvDisqualifyForEhWriteThru)
+        {
+            if (tree->gtFlags & GTF_VAR_DEF)
+            {
+                bool bbInALoop             = (block->bbFlags & BBF_BACKWARD_JUMP) != 0;
+                bool bbIsReturn            = block->bbJumpKind == BBJ_RETURN;
+                bool needsExplicitZeroInit = fgVarNeedsExplicitZeroInit(lclNum, bbInALoop, bbIsReturn);
+
+                if ((varDsc->lvEhWriteThruCandidate == true) || (needsExplicitZeroInit == true) ||
+                    (tree->gtFlags & GTF_COLON_COND) || (tree->gtFlags & GTF_VAR_USEASG))
```

Comment:
Maybe add comment on why these cause exclusion?

`USEASG` I can guess, at though it seems it would be redundant with restricting to single def?

`COLON_COND` is a mystery to me, why does it matter? It goes away during morph.

Reply:
To be honest, I just copied the conditions of `USEASG` and `COLON_COND` from `lvSingleDef` logic above. Yes, I don't think we need it specially because we already check for `varDsc->lvEhWriteThruCandidate`.